src/server.ts

```diff
@@ -1,331 +1,95 @@
-    const alreadyReacted = (reaction.count && reaction.count > 1) == true;
-
-    //dangerous - allow seabot to react to a bot's commands for creating server events
-    if (
-        !alreadyReacted &&
-        message.author?.id === UserIDs.APOLLO &&
-        emoji.name === createServerEvent.emojiId &&
-        isModReaction(reaction, user) && 
-        !Environment.DEBUG
-    ) {
-        const event = await createServerEvent.execute(message) as GuildScheduledEvent;
-        try{ 
-            if(event){
-                //confirm event creation
-                await message.react('✅');
-                //attempt to message user who created it with link
-                await user.send(`Event created: ${event.url}`);
-            }
-        }
-        catch(e: any) {
-            console.dir(e);
-        }
-        finally{
-            //stop processing  reactions after this
-            return;
-        }
+async function startDiscordBot() {
+    console.log("Starting bot...");
+    try {
+        const eventRouter = new DiscordEventRouter(discordBot.client);
+        eventRouter.addEventListener(Events.InteractionCreate, processModReportInteractions);
+        eventRouter.addEventListener(Events.VoiceStateUpdate, handleVoiceStatusUpdate);
+        eventRouter.addEventListener(Events.ClientReady, announcePresence);
+        eventRouter.addEventListener(Events.ClientReady, startCronJobs);
+
+        await discordBot.start(eventRouter);
+    } catch (error) {
+        console.error("Fatal error while starting bot:");
+        console.error(error);
+        exit(1);
     }
+}
 
-    // this prevents the same reaction command from firing multiple times
-    if(message.author?.bot || !emoji.id || alreadyReacted) {
-        return;
+function startExpressServer() {
+    try {
+        expressServer.start();
+    } catch (error) {
+        console.error("Fatal error while starting Express server:");
+        console.error(error);
+        exit(1);
     }
-    try{
-        const command = reactionCommands?.[emoji.id];
-        command?.execute(message);
-        if(command?.removeReaction) {
-            await reaction.remove();
-        }
-    }
-    catch(e: any){
-        console.dir(e);
-        message.react('💩')
-    }
-});
-
-//process slash commands
-client.on("interactionCreate", async interaction => {
-    if(!interaction.isChatInputCommand()) return;
-
-    const command = commands?.[interaction.commandName];
-    if(command) {
-        command.executeSlashCommand?.(interaction);
-    };
-});
-
-// handle mod report reactions
-client.on('interactionCreate', processModReportInteractions);
+}
 
-//handle voice connections
-client.on('voiceStateUpdate', handleVoiceStatusUpdate);
-//join/leave
-client.on('guildMemberRemove', abeLeaves);
-client.on('guildMemberAdd', newAccountJoins);
-// #endregion
+async function loadConfiguration(path: string): Promise<ISeabotConfig> {
+    path = `${path}\\seabotConfig.json`;
```

Comment:
Just a note (for me more than anything) that using a config file at a specific path is going to require a minor change to how we handle deploying the config and the bot itself. I'm also working on some changes in the CI/CD pipelines and general release process so I'll keep this in mind.

Reply:
It actually loads from the root of the folder the script is running from currently (__dirname). I left the implementation open to load from other paths, in case of wanting to load a fallback config file, or loading different configs based on environment, etc.